Alameen688/mydiary-spa
# #162190554 User should be able to sign up

#### What does this PR do?
- Implements user registration
#### Description of Task to be completed?
- Add proper validation
- Allow user with valid input values and no existing account to sign up

#### How should this be manually tested?
- Run `npm run dev`
- Go to [localhost:8080/signup](http://localhost:8080/signup)
- Create an account. You should be taken to the login page if signup was successful.

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?

[#162190554](https://www.pivotaltracker.com/story/show/{162190554})

#### Screenshots (if appropriate)
![screen shot 2018-11-29 at 8 32 40 pm 2](https://user-images.githubusercontent.com/9039613/49246994-78ec2580-f416-11e8-8e73-842ed46b9b24.png)


#### Questions:
